### PR TITLE
Fcma mvpa searchlight

### PR DESCRIPTION
**Pull request by yidawang**
enable activity-based voxel selection in the FCMA pipeline

### CODE REVIEW

File: examples/fcma/file_io.py [hunk 1]
```diff
@@ -178,3 +178,54 @@ def prepare_data(data_dir, extension, mask_file, epoch_file):
...
+        assuming all subjects have the same number of epochs
+        len(epoch_list) equals the number of subjects
+        also assuming an epoch is always a continuous time course
+
+    Returns
+    -------
+    epoch\_info: list of tuple (label, sid, start, end)
+        the condition labels of the epochs
+        len(labels) labels equals the number of epochs
+        assuming the epochs of the same sid are adjacent
```
Comment (mihaic) [inline examples/fcma/file_io.py]: Please write sentences in the docstrings. They are hard to follow without capitalization and punctuation.

File: examples/fcma/file_io.py [hunk 2]
```diff
@@ -178,3 +178,54 @@ def prepare_data(data_dir, extension, mask_file, epoch_file):
...
+            for eid in range(epoch.shape[1]):
+                r = np.sum(sub_epoch[eid, :])
+                if r > 0:   # there is an epoch in this condition
+                    start = np.nonzero(sub_epoch[eid, :])[0][0]
+                    epoch_info.append((cond, sid, start, start+r))
+    time2 = time.time()
+    logger.info(
+        'epoch separation done, takes %.2f s' %
+        (time2 - time1)
+    )
```
Comment (mihaic) [inline examples/fcma/file_io.py]: Isn't `debug` more appropriate for timing messages?

File: examples/fcma/mvpa_voxel_selection.py [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+# if want to output log to a file instead of outputting log to the console,
+# replace "stream=sys.stdout" with "filename='fcma.log'"
+logging.basicConfig(level=logging.INFO, format=format, stream=sys.stdout)
+logger = logging.getLogger(__name__)
+
+"""
+example running command:
+mpirun -np 2 python mvpa_voxel_selection.py /Users/yidawang/data/face_scene/raw nii.gz /Users/yidawang/data/face_scene/mask.nii.gz
+                        data/fs_epoch_labels.npy 18
+"""
```
Comment (mihaic) [inline examples/fcma/mvpa_voxel_selection.py]: How about providing some sample data like in SRM?
https://github.com/IntelPNI/brainiak/blob/master/examples/funcalign/download-data.sh

If the files are small enough (a few kB), you can even add them to Git.

Comment (mihaic) [inline examples/fcma/mvpa_voxel_selection.py]: Also, how about making a shell script with the command, like for HTFA?
https://github.com/IntelPNI/brainiak/blob/master/examples/factoranalysis/run_mpi_htfa.sh

File: brainiak/fcma/mvpa_voxelselector.py [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+                       in the list
+                    5. voxel selection is always done in the auto-correlation,
+                       i.e. raw_data correlate with themselves
+                    6. if MPI jobs are running on multiple nodes, the user
+                       home directory is shared by all nodes
+
+    mask: 3D array
+
+    epoch\_info: list of tuple (label, sid, start, end)
+        the condition labels of the epochs
```
Comment (yidawang) [inline brainiak/fcma/mvpa_voxelselector.py]: fixed

File: brainiak/fcma/mvpa_voxelselector.py [hunk 5]
```diff
@@ -0,0 +1,195 @@
...
+        len(labels) labels equals the number of epochs
+        assuming the epochs of the same sid are adjacent,
+        and sid is the index of raw\_data
```
Comment (yidawang) [inline brainiak/fcma/mvpa_voxelselector.py]: done

File: brainiak/fcma/mvpa_voxelselector.py [hunk 6]
```diff
@@ -0,0 +1,195 @@
...
+
+    num\_folds: int
+        the number of folds to be conducted in the cross validation
+
+    sl: Searchlight
+        the distributed Searchlight object
+
+    processed\_data: 4D array in shape [brain 3D + epoch]
```
Comment (yidawang) [inline brainiak/fcma/mvpa_voxelselector.py]: The parameter list here documents the properties of `MVPAVoxelSelector` instead of the input of `__init__`. The two voxel selector of FCMA is not about fitting, so I didn't follow the standard to only pass data through `fit`. I agree to keep the code structure as it is and leave the restructure discussion later. For now, let me modify the properties not passed through `__init__` as `var_`.

File: brainiak/fcma/mvpa_voxelselector.py [hunk 7]
```diff
@@ -0,0 +1,195 @@
...
+        contains the averaged and normalized brain data epoch by epoch
+        it is generated in _\preprocess\_data method
```
Comment (yidawang) [inline brainiak/fcma/mvpa_voxelselector.py]: fixed

File: brainiak/fcma/mvpa_voxelselector.py [hunk 8]
```diff
@@ -0,0 +1,195 @@
...
+                 epoch_info,
+                 num_folds,
+                 sl_rad=2,
+                 max_blk_edge=10
+                 ):
+        self.raw_data = raw_data
+        self.mask = mask.astype(np.bool)
+        self.epoch_info = epoch_info
+        self.num_folds = num_folds
+        self.sl = Searchlight(sl_rad=sl_rad, max_blk_edge=max_blk_edge)
```
Comment (yidawang) [inline brainiak/fcma/mvpa_voxelselector.py]: done

File: examples/fcma/file_io.py (hunk 1 above)
Comment (yidawang) [inline examples/fcma/file_io.py]: done

File: examples/fcma/file_io.py (hunk 2 above)
Comment (yidawang) [inline examples/fcma/file_io.py]: done

File: examples/fcma/mvpa_voxel_selection.py (hunk 3 above)
Comment (yidawang) [inline examples/fcma/mvpa_voxel_selection.py]: good idea, done
